### PR TITLE
Cdk Pattern C#

### PR DESCRIPTION
**Pull request by andrestoll**
Added CDK project with
- single resources for SAR application

Note:
README only shows deployment steps, not sure if we need to whole README for cdk patterns

### CODE REVIEW

File: cdk/csharp/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# How to deploy the AWS Lambda Power Tuner tool using the CDK for C#
+
+This CDK project deploys the *AWS Lambda Power Tuner* tool.
+
+You can use the project as a standalone or reuse it within your own CDK projects.
+
+## Prerequisites
+
+- [AWS CDK Toolkit](https://docs.aws.amazon.com/cdk/v2/guide/getting_started.html#getting_started_install)
+- [.NET 8.0 or later](https://docs.aws.amazon.com/cdk/v2/guide/getting_started.html#getting_started_prerequisites)
```
Comment (andrestoll) [inline cdk/csharp/README.md:10]: I specified .NET 8.0 here, also used in projects (CDK guide says .NET 6.0)

Comment (alexcasalboni) [reply cdk/csharp/README.md:10]: @andrestoll @Anirudh1612 I'm wondering if we should standardize on one single README for all the CDK sub-folders? 

See the TypeScript PR here too: https://github.com/alexcasalboni/aws-lambda-power-tuning/pull/230

If it does make sense to keep individual README files for each language, how do we make each one useful with language-specific tips or guidance?

File: cdk/csharp/cdk.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      "README.md",
+      "cdk*.json",
+      "src/*/obj",
+      "src/*/bin",
+      "src/*.sln",
+      "src/*/GlobalSuppressions.cs",
+      "src/*/*.csproj"
+    ]
+  },
+  "context": {
```
Comment (alexcasalboni) [inline cdk/csharp/cdk.json:17]: Similar comment here: do we need all of these for this CDK project? Is this a common default or just a copy-n-past from another project?

File: cdk/csharp/src/Cdk.sln [hunk 3]
```diff
@@ -0,0 +1,48 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio 15
+VisualStudioVersion = 15.0.26124.0
+MinimumVisualStudioVersion = 15.0.26124.0
```
Comment (alexcasalboni) [inline cdk/csharp/src/Cdk.sln:5]: Is this a common value? Is it worth documenting in the README?

I have no idea about .NET development best practices 😄

File: cdk/csharp/src/src/Cdk/Cdk.csproj [hunk 4]
```diff
@@ -0,0 +1,25 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (alexcasalboni) [inline cdk/csharp/src/src/Cdk/Cdk.csproj:1]: Is the double `/src/src/` folder a .NET/VisualStudio convention?

File: cdk/csharp/src/src/Cdk/TheLambdaPowerTunerStack.cs [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+{
+    public class TheLambdaPowerTunerStack : Stack
+    {
+        internal TheLambdaPowerTunerStack(Construct scope, string id, IStackProps props = null) : base(scope, id, props)
+        {
+            var stateMachineConfiguration = new Dictionary<string, string>
+            {
+                { "lambdaResource", "*" },
+                { "PowerValues", "128,256,512,1024,1536,3008" },
+                { "visualizationURL", "https://lambda-power-tuning.show/" },
```
Comment (alexcasalboni) [inline cdk/csharp/src/src/Cdk/TheLambdaPowerTunerStack.cs:16]: This is already the default value - I'd comment it out (like the others below)

File: cdk/csharp/src/src/Cdk/TheLambdaPowerTunerStack.cs [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+                { "totalExecutionTimeout", "300" },
+                { "payloadS3Key", "*" },
```
Comment (alexcasalboni) [inline cdk/csharp/src/src/Cdk/TheLambdaPowerTunerStack.cs:18]: Same here - I'd comment it out

File: cdk/csharp/src/src/Cdk/TheLambdaPowerTunerStack.cs [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    public class TheLambdaPowerTunerStack : Stack
+    {
+        internal TheLambdaPowerTunerStack(Construct scope, string id, IStackProps props = null) : base(scope, id, props)
+        {
+            var stateMachineConfiguration = new Dictionary<string, string>
+            {
+                { "lambdaResource", "*" },
+                { "PowerValues", "128,256,512,1024,1536,3008" },
+                { "visualizationURL", "https://lambda-power-tuning.show/" },
+                { "totalExecutionTimeout", "300" },
```
Comment (alexcasalboni) [inline cdk/csharp/src/src/Cdk/TheLambdaPowerTunerStack.cs:17]: Same here - I'd comment it out

File: cdk/csharp/src/src/Cdk/TheLambdaPowerTunerStack.cs [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+namespace TheLambdaPowerTunerStack
+{
+    public class TheLambdaPowerTunerStack : Stack
+    {
+        internal TheLambdaPowerTunerStack(Construct scope, string id, IStackProps props = null) : base(scope, id, props)
+        {
+            var stateMachineConfiguration = new Dictionary<string, string>
+            {
+                { "lambdaResource", "*" },
```
Comment (alexcasalboni) [inline cdk/csharp/src/src/Cdk/TheLambdaPowerTunerStack.cs:14]: Same here - I'd comment it out

File: cdk/csharp/src/src/Cdk/TheLambdaPowerTunerStack.cs [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+                { "PowerValues", "128,256,512,1024,1536,3008" },
```
Comment (alexcasalboni) [inline cdk/csharp/src/src/Cdk/TheLambdaPowerTunerStack.cs:15]: Same here - I'd comment it out

File: cdk/csharp/src/src/Cdk/TheLambdaPowerTunerStack.cs [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+                // { "layerSdkName", "<name of the SDK layer>" },
+                // { "securityGroupIds", "<List of Security Groups to use in every Lambda function's VPC Configuration>" },
+                // { "subnetIds", "<List of Subnets to use in every Lambda function's VPC Configuration>" },
+            };
+
+            new CfnApplication(this, "SAR", new CfnApplicationProps
+                {
+                    Location = new CfnApplication.ApplicationLocationProperty {
+                        ApplicationId = "arn:aws:serverlessrepo:us-east-1:451282441545:applications/aws-lambda-power-tuning",
+                        SemanticVersion = "4.3.0"
```
Comment (alexcasalboni) [inline cdk/csharp/src/src/Cdk/TheLambdaPowerTunerStack.cs:32]: Please use the latest version here -> `4.3.4`

**Review by alexcasalboni — CHANGES_REQUESTED** [review]
Thanks for contributing 🙏  A few comments to address :)

File: cdk/csharp/src/Cdk.sln (hunk 3 above)
Comment (andrestoll) [reply cdk/csharp/src/Cdk.sln:5]: This is a standard solution file defining the projects. 
All common values, see here : https://learn.microsoft.com/en-us/visualstudio/extensibility/internals/solution-dot-sln-file?view=vs-2022#file-header
I don't think we need to document this.

File: cdk/csharp/src/src/Cdk/Cdk.csproj (hunk 4 above)
Comment (andrestoll) [reply cdk/csharp/src/src/Cdk/Cdk.csproj:1]: no actually not, but one usually has a src folder containing the projects and a tests folder along side it (.NET project setup).
The parent src came from the cdk init command. 
I propose we change to:
```
/csharp
  /src
    /Cdk
  /tests
    /Cdk.Tests
  .gitignore
  README.md (depending on above comment)
  cdk.json
  Cdk.sln
```

Comment (alexcasalboni) [reply cdk/csharp/src/src/Cdk/Cdk.csproj:1]: @andrestoll sounds good!

File: cdk/csharp/src/src/Cdk/TheLambdaPowerTunerStack.cs (hunk 9 above)
Comment (andrestoll) [reply cdk/csharp/src/src/Cdk/TheLambdaPowerTunerStack.cs]: done

File: cdk/csharp/src/src/Cdk/TheLambdaPowerTunerStack.cs (hunk 8 above)
Comment (andrestoll) [reply cdk/csharp/src/src/Cdk/TheLambdaPowerTunerStack.cs]: done

**Review by alexcasalboni — APPROVED** [review]
Ready to merge 🎉

**Pull request merged**